### PR TITLE
lint for `print()` statements

### PR DESCRIPTION
**Pull request by ThomasHepworth**
### Type of PR

- [x] MAINT

Quick PR to add the [flake8-print](https://beta.ruff.rs/docs/rules/#flake8-print-t20) linter to repo. This should catch any pointless `print()` statements we forget to remove (excl. those that are commented).

### CODE REVIEW

**Comment by ThomasHepworth** [comment]
@ADBond can you give the new docs a 5 min check?

File: docs/dev_guides/changing_splink/lint_and_format.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+## Linting your code
+
+For linting, we currently make use of [ruff](https://github.com/charliermarsh/ruff) to perform checks, and if desired, *some* fixes for your data.
```
Comment (ADBond) [inline docs/dev_guides/changing_splink/lint_and_format.md:3]: 'data' -> 'code' ?

**Review by ADBond — APPROVED** [review]
Nice, think those are some good additions 👍

File: docs/dev_guides/changing_splink/lint_and_format.md (hunk 1 above)
Comment (ThomasHepworth) [reply docs/dev_guides/changing_splink/lint_and_format.md:3]: Ah, good spot.

Comment (ThomasHepworth) [inline docs/dev_guides/changing_splink/lint_and_format.md:3]:
```suggestion
For linting, we currently make use of [ruff](https://github.com/charliermarsh/ruff) to perform checks, and if desired, *some* fixes for your code.
```

**Pull request merged**